pkg/agent/svid/rotator_test.go

```diff
@@ -89,41 +191,64 @@ func TestRotator(t *testing.T) {
 
 			// Subscribe to SVID changes and run the rotator
 			stream := rotator.Subscribe()
-
 			ctx, cancel := context.WithCancel(context.Background())
 			errCh := make(chan error, 1)
 			go func() {
 				errCh <- rotator.Run(ctx)
 			}()
 
-			// Wait for the initial rotation check to finish
+			// All tests should get through one rotation loop or error
 			select {
-			case <-time.After(time.Minute):
-				t.Fatal("timed out waiting for rotation check to finish")
-			case <-rotationDone:
-				clk.WaitForAfter(time.Minute, "timed out waiting for rotation loop to start waiting")
+			case <-clk.WaitForAfterCh():
+			case err = <-errCh:
+				if tt.failTest {
+					cancel()
+					spiretest.RequireErrorContains(t, err, tt.err)
+					return
+				}
+
+				t.Fatalf("unexpected error during first rotation loop: %v", err)
+			case <-time.After(testTimeout):
+				if hook.LastEntry() != nil && hook.LastEntry().Level == logrus.ErrorLevel {
+					t.Fatalf("timed out waiting for first rotation loop to finish: %s", hook.LastEntry().Message)
+				}
+				t.Fatal("timed out waiting for first rotation loop to finish")
 			}
 
-			// Optionally advance the clock by the specified amount
-			// and wait for another rotation check to finish.
-			if tt.checkAfter != 0 {
-				require.Greater(t, tt.checkAfter, DefaultRotatorInterval, "checkAfter should be larger than the default rotator interval")
-				clk.Add(tt.checkAfter)
+			// Wait for the rotation check to finish
+			if tt.shouldRotate {
+				// Optionally advance the clock by the specified amount
+				// before waiting for the rotation check to finish.
+				if tt.notAfter != 0 {
+					require.Greaterf(t, tt.notAfter, DefaultRotatorInterval, "notAfter must be larger than %v", DefaultRotatorInterval)
+					clk.Add(tt.notAfter)
+				}
+
 				select {
-				case <-time.After(time.Minute):
-					t.Fatal("timed out waiting for rotation check to finish after clock adjustment")
 				case <-rotationDone:
-					clk.WaitForAfter(time.Minute, "timed out waiting for rotation loop to start waiting after clock adjustment")
+				case err = <-errCh:
+					if tt.failTest {
```

Comment:
It is strange to have the same var to be used on different places,  one is for failing for should rotate and another for reattestation? maybe you can create different vars for each of them?

Reply:
done